app/addons/auth/actions.js

```diff
@@ -10,148 +10,162 @@
 }
 
-function updateChangePasswordField (value) {
+export function updateChangePasswordField(value) {
   FauxtonAPI.dispatch({
-    type: ActionTypes.AUTH_UPDATE_CHANGE_PWD_FIELD,
+    type: AUTH_UPDATE_CHANGE_PWD_FIELD,
     value: value
   });
 }
 
-function updateChangePasswordConfirmField (value) {
+export function updateChangePasswordConfirmField(value) {
   FauxtonAPI.dispatch({
-    type: ActionTypes.AUTH_UPDATE_CHANGE_PWD_CONFIRM_FIELD,
+    type: AUTH_UPDATE_CHANGE_PWD_CONFIRM_FIELD,
     value: value
   });
 }
 
-function createAdmin (username, password, loginAfter) {
+export function createAdmin(username, password, loginAfter) {
   var nodes = nodesStore.getNodes();
-  var promise = FauxtonAPI.session.createAdmin(username, password, loginAfter, nodes[0].node);
-
-  promise.then(() => {
-    FauxtonAPI.addNotification({ msg: FauxtonAPI.session.messages.adminCreated });
-    if (loginAfter) {
-      FauxtonAPI.navigate('/');
-    } else {
-      FauxtonAPI.dispatch({ type: ActionTypes.AUTH_CLEAR_CREATE_ADMIN_FIELDS });
+  FauxtonAPI.session.createAdmin(
+    username,
+    password,
+    loginAfter,
+    nodes[0].node
+  )
+  .then(
+    () => {
+      FauxtonAPI.addNotification({
+        msg: FauxtonAPI.session.messages.adminCreated
+      });
+      if (loginAfter) {
+        FauxtonAPI.navigate("/");
+      } else {
+        FauxtonAPI.dispatch({ type: AUTH_CLEAR_CREATE_ADMIN_FIELDS });
+      }
+    },
+    (xhr, type, msg) => {
+      msg = xhr;
+      if (arguments.length === 3) {
+        msg = xhr.responseJSON.reason;
+      }
+      errorHandler(
+        FauxtonAPI.session.messages.adminCreationFailedPrefix + " " + msg
+      );
     }
-  }, (xhr, type, msg) => {
-    msg = xhr;
-    if (arguments.length === 3) {
-      msg = xhr.responseJSON.reason;
-    }
-    errorHandler(FauxtonAPI.session.messages.adminCreationFailedPrefix + ' ' + msg);
-  });
+  );
 }
 
 // simple authentication method - does nothing other than check creds
-function authenticate (username, password, onSuccess) {
-  $.ajax({
-    cache: false,
-    type: 'POST',
-    url: '/_session',
-    dataType: 'json',
-    data: { name: username, password: password }
-  }).then(() => {
-    FauxtonAPI.dispatch({
-      type: ActionTypes.AUTH_CREDS_VALID,
-      options: { username: username, password: password }
-    });
-    hidePasswordModal();
-    onSuccess(username, password);
-  }, () => {
-    FauxtonAPI.addNotification({
-      msg: 'Your password is incorrect.',
-      type: 'error',
-      clear: true
-    });
-    FauxtonAPI.dispatch({
-      type: ActionTypes.AUTH_CREDS_INVALID,
-      options: { username: username, password: password }
-    });
-  });
+export function authenticate(username, password, onSuccess) {
+  session
+    .create({
+      name: username,
+      password: password
+    })
+    .then(
+      () => {
+        FauxtonAPI.dispatch({
+          type: AUTH_CREDS_VALID,
+          options: { username: username, password: password }
```

Comment:
With es2015 you can make this options object look like this

```
{
 usernane,
password
}
```

Could you check if other objects could use that format as well.